sdk/synapse/Azure.Analytics.Synapse.Spark/src/Customization/SparkBatchOperation.cs

```diff
@@ -0,0 +1,126 @@
+        private bool _completed;
+
+        internal SparkBatchOperation(SparkBatchClient client, ClientDiagnostics diagnostics, Response<SparkBatchJob> response)
+        {
+            _client = client;
+            _value = response.Value ?? throw new InvalidOperationException("The response does not contain a value.");
+            _response = response;
+            _completed = false;
+            _diagnostics = diagnostics;
+        }
+
+        /// <inheritdoc/>
+        public override string Id => _value.Id.ToString(CultureInfo.InvariantCulture);
+
+        /// <summary>
+        /// Gets the <see cref="SparkBatchJob"/>.
+        /// You should await <see cref="WaitForCompletionAsync(CancellationToken)"/> before attempting to use session in this pending state.
+        /// </summary>
+        /// <remarks>
+        /// Azure Synapse will return a <see cref="SparkBatchJob"/> immediately but may take time to the session to be ready.
+        /// </remarks>
+        public override SparkBatchJob Value => _value;
+
+        /// <inheritdoc/>
+        public override bool HasCompleted => _completed;
+
+        /// <inheritdoc/>
+        public override bool HasValue => true;
+
+        /// <inheritdoc/>
+        public override Response GetRawResponse() => _response.GetRawResponse();
+
+        /// <inheritdoc/>
+        public override Response UpdateStatus(CancellationToken cancellationToken = default) => UpdateStatusAsync(false, cancellationToken).EnsureCompleted();
+
+        /// <inheritdoc/>
+        public override async ValueTask<Response> UpdateStatusAsync(CancellationToken cancellationToken = default) => await UpdateStatusAsync(true, cancellationToken).ConfigureAwait(false);
+
+        /// <inheritdoc />
+        public override ValueTask<Response<SparkBatchJob>> WaitForCompletionAsync(CancellationToken cancellationToken = default) =>
+            this.DefaultWaitForCompletionAsync(s_defaultPollingInterval, cancellationToken);
+
+        /// <inheritdoc />
+        public override ValueTask<Response<SparkBatchJob>> WaitForCompletionAsync(TimeSpan pollingInterval, CancellationToken cancellationToken) =>
+            this.DefaultWaitForCompletionAsync(pollingInterval, cancellationToken);
+
+        private async ValueTask<Response> UpdateStatusAsync(bool async, CancellationToken cancellationToken)
+        {
+            if (!_completed)
+            {
+                using DiagnosticScope scope = _diagnostics.CreateScope($"{nameof(SparkSessionOperation)}.{nameof(UpdateStatus)}");
+                scope.Start();
+
+                try
+                {
+                    if (async)
+                    {
+                        _response = await _client.GetSparkBatchJobAsync(_value.Id, true, cancellationToken).ConfigureAwait(false);
+                    }
+                    else
+                    {
+                        _response = _client.GetSparkBatchJob(_value.Id, true, cancellationToken);
```

Comment:
Wouldn't we need to update the _value also since we store it separately? Might be better to just grab the value from _response using the implicit cast when necessary.

Reply:
The ID does not change, but let me refactor to store that instead.